src/System.Collections.NonGeneric/tests/Hashtable/PropertySyncRootTests.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.Threading;
+using System.Collections;
+using System.Threading.Tasks;
+
+namespace System.Collections.HashtableTests
+{
+    public class SyncRootTests
+    {
+        private Hashtable _arrDaughter;
+        private Hashtable _arrGrandDaughter;
+        private Int32 _iNumberOfElements = 100;
+
+        [Fact]
+        public void TestGetSyncRootBasic()
+        {
+            Hashtable arrSon;
+            Hashtable arrMother;
+
+            Task[] workers;
+            Action ts1;
+            Action ts2;
+            Int32 iNumberOfWorkers = 30;
+            Hashtable hshPossibleValues;
+            IDictionaryEnumerator idic;
+
+            Hashtable hsh1;
+            Hashtable hsh2;
+            Hashtable hsh3;
+
+            //[] we will create different HT and make sure that they return different SyncRoot values
+            hsh1 = new Hashtable();
+            hsh2 = new Hashtable();
+
+            Assert.NotEqual(hsh1.SyncRoot, hsh2.SyncRoot);
+            Assert.Equal(hsh1.SyncRoot.GetType(), typeof(object));
+
```

Comment:
Is it important/documented that Hashtable.SyncRoot.GetType() == typeof(object)?

The public documentation says (table.SyncRoot is object) is true, but not that it won't be some unspecified derived type.  I'm just not sure why this is being checked.

(Reference: https://msdn.microsoft.com/en-us/library/system.collections.hashtable.syncroot(v=vs.110).aspx)